src/chunk.writer.cpp

```diff
@@ -210,13 +219,26 @@ ChunkWriter::open_chunk_file()
     std::string path = (fs::path(base_dir_) / file_path).string();
     auto parent_path = fs::path(path).parent_path();
 
-    if (!fs::is_directory(parent_path))
-        fs::create_directories(parent_path);
+    if (!fs::is_directory(parent_path)) {
+        std::error_code ec;
+        // the call may fail if the directory is being created in another thread
+        if (!fs::create_directories(parent_path, ec) &&
+            !fs::is_directory(parent_path)) {
+            LOGE("Failed to create directory '%s': %s",
+                 parent_path.string().c_str(),
+                 ec.message().c_str());
+            return false;
```

Comment:
I like `create_directories`. 

This looks like it fails if another thread is creating the folder at the same time. What happens then? How does it recover?

Reply:
If I understood correctly, then we'll skip this chunk in the failing case. I don't know if/how zarr readers deal with missing chunks, but I guess we don't want to have missing chunks if we can avoid it. I guess we could retry, but I guess that will complicate the scheduling/ordering logic.

Another thought is to guard calls to `create_directories` with a lock, but that obviously comes with a performance hit that I can't really analyze without more data.

Or could we create all the directories we need on the main scheduling thread before we start writing the chunks?

Reply:
> failure in a thread should be a fatal error

Yes. Any failure to record data should stop the acquisition.

Reply:
There are a couple approaches I can imagine here. 

One is to retry for a while until success. The idea is that this block has to ensure a writable directory for the next part of the code. There should be an associated timeout.

Two is to create a unique temporary directory and then move it later? That seems like it can't work but I thought I'd throw it out there.